src/sync/app.cpp

```diff
@@ -841,5 +842,53 @@ void App::link_user(std::shared_ptr<SyncUser> user,
     }
 
 
+RemoteMongoClient App::remote_mongo_client(const std::string& service_name)
+{
+    m_service_name = util::Optional<std::string>(service_name);
+    RemoteMongoClient remote_client(std::make_unique<App>(*this));
+    return remote_client;
+}
+
+void App::call_function(const std::string& name,
+                        const std::string& args_json,
+                        std::function<void (util::Optional<AppError>, util::Optional<std::string>)> completion_block) const
+{
+    call_function(name, args_json,
+                  util::Optional<std::string>(m_service_name),
+                  completion_block);
+}
+
+void App::call_function(const std::string& name,
+                   const std::string& args_json,
+                   const util::Optional<std::string>& service_name,
+                   std::function<void (util::Optional<AppError>, util::Optional<std::string>)> completion_block) const
+{
+    auto handler = [completion_block](const Response& response) {
+
+        if (auto error = check_for_errors(response)) {
+            return completion_block(error, util::none);
+        }
+        
+        completion_block(util::none, util::Optional<std::string>(response.body));
+    };
+
+    std::string route = util::format("%1/app/%2/functions/call", m_base_route, m_config.app_id);
+
+    auto args = nlohmann::json::parse(args_json);
+    args.push_back({ "name" , name });
+    if (service_name) {
+        args.push_back({ "service" , service_name.value() });
```

Comment:
Getting the value out of an optional is normally done by dereferencing it (i.e. `*service_name`) rather than calling `value()`.

Reply:
changed to use dereferencing